### PR TITLE
Add menu item and hotkey to center view on current selection

### PR DESCRIPTION
**Pull request by mlfarrell**
## Prerequisites
- [x] needs https://github.com/KittyCAD/engine/pull/2670 merged before this can merge.
- [x] needs a bump of `@kittycad/lib`

Adds a new modelingMachine event and action to send the new `default_camera_center_to_selection` engine command, and connects this event to the `mod + shift + c` hotkey and the view menu in the lower-right view controls

### CODE REVIEW

File: src/components/ModelingMachineProvider.tsx [hunk 1]
```diff
@@ -1037,6 +1038,21 @@ export const ModelingMachineProvider = ({
     modelingSend({ type: 'Delete selection' })
   })
 
+  // Allow ctrl+alt+c to center to selection
+  useHotkeys(['ctrl + alt + c'], () => {
```
Comment (jtran) [inline src/components/ModelingMachineProvider.tsx:1042]: `mod` gets translated to Control on Windows and Linux and Command on Mac.

```suggestion
  useHotkeys(['mod + alt + c'], () => {
```

**Comment by franknoirot** [comment]
I'm tagging in to do the following:

- [ ] add this hotkey to the list of fixed hotkeys shown in the dialog
- [ ] fmting and clean up (remove the log)
- [ ] add an E2E test

**Comment by mlfarrell** [comment]
my b on the log.  i forgot to remove that

**Comment by franknoirot** [comment]
Oh it's nothing at all, not to worry

**Comment by franknoirot** [comment]
I think the last thing this needs is a bump to the `@kittycad/lib` so that `tsc` doesn't yell at us for using the newfangled `default_camera_center_to_selection` command

**Comment by franknoirot** [comment]
@jtran do you mind reviewing this? I don't think I should since I contributed to it

**Comment by lf94** [comment]
Oh shit yes, first thing in the morning!

File: e2e/playwright/fixtures/sceneFixture.ts [hunk 2]
```diff
@@ -87,6 +110,36 @@ export class SceneFixture {
     )
     await closeDebugPanel(this.page)
   }
+  /** Forces a refresh of the camera position and target displayed
+   *  in the debug panel and then returns the values of the fields
+   */
+  async getCameraInfo() {
```
Comment (lf94) [inline e2e/playwright/fixtures/sceneFixture.ts:116]: I love our new fixture stuff :')

File: e2e/playwright/testing-gizmo.spec.ts [hunk 3]
```diff
@@ -1,18 +1,18 @@
-import { test, expect } from '@playwright/test'
-
+import { _test, _expect } from './playwright-deprecated'
```
Comment (lf94) [inline e2e/playwright/testing-gizmo.spec.ts:1]: I mustve missed the memo - what's going on here? :)

File: src/components/ModelingMachineProvider.tsx [hunk 4]
```diff
@@ -243,6 +244,17 @@ export const ModelingMachineProvider = ({
...
+        'Center camera on selection': () => {
+          engineCommandManager
+            .sendSceneCommand({
+              type: 'modeling_cmd_req',
+              cmd_id: uuidv4(),
+              cmd: {
+                type: 'default_camera_center_to_selection',
+              },
+            })
+            .catch(reportRejection)
```
Comment (lf94) [inline src/components/ModelingMachineProvider.tsx:256]: IMO should be in the CameraControls file but we can refactor later!

File: e2e/playwright/testing-gizmo.spec.ts [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
+import { test } from './fixtures/fixtureSetup'
```
Comment (nadr0) [inline e2e/playwright/testing-gizmo.spec.ts:2]: Can we revert this name change and do something like

```
import { test, expect } from '@playwright/test'
import { test as fixtureSetupTest } from './fixtures/fixtureSetup'
```

I am confused to why we have a `'./playwright-deprecated'` when we can just do alias importing?

File: src/components/ModelingMachineProvider.tsx (hunk 4 above)
Comment (franknoirot) [reply src/components/ModelingMachineProvider.tsx:256]: Oh yeah good idea, that's better

File: e2e/playwright/testing-gizmo.spec.ts (hunk 3 above)
Comment (franknoirot) [reply e2e/playwright/testing-gizmo.spec.ts:1]: We'll undo this change in a separate PR per Slack discussion

File: src/clientSideScene/CameraControls.ts [hunk 6]
```diff
@@ -893,6 +893,7 @@ export class CameraControls {
         type: 'zoom_to_fit',
         object_ids: [], // leave empty to zoom to all objects
         padding: 0.2, // padding around the objects
+        animated: false, // don't animate the zoom for now
```
[the remark this answers is not in the document]
Comment (franknoirot) [reply src/clientSideScene/CameraControls.ts:896]: These weren't animated before, but the types got updated to make the boolean required in the object. I tried out setting it to `true` but the user experience was not good; I'd need to understand how to use that flag better before turning it on.

**Pull request merged**